src/main/java/dev/compactmods/machines/util/CompactStructureGenerator.java

```diff
@@ -73,26 +77,14 @@ public static void generateCompactWall(LevelAccessor world, RoomSize size, Block
      * @param center
      */
     public static void generateCompactStructure(LevelAccessor world, RoomSize size, BlockPos center) {
-        int s = size.getInternalSize() / 2;
-
-        BlockPos floorCenter = center.relative(Direction.DOWN, s);
-        BlockPos machineTopCenter = center.relative(Direction.UP, s);
-
-        AABB floorBlocks = new AABB(floorCenter, floorCenter)
-                .inflate(s, 0, s);
-        AABB machineInternal = new AABB(machineTopCenter, floorCenter)
-                .inflate(s, 0, s);
-
-
-        boolean anyAir = BlockPos.betweenClosedStream(floorBlocks).anyMatch(world::isEmptyBlock);
-
-        if (anyAir) {            // Generate the walls
-            Arrays.stream(Direction.values())
-                    .forEach(d -> generateCompactWall(world, size, center, d));
-
-            BlockPos.betweenClosedStream(machineInternal)
-                    .forEach(p -> world.setBlock(p, Blocks.AIR.defaultBlockState(), 7));
-
+        world.setBlock(center, Blocks.STRUCTURE_BLOCK.defaultBlockState(), Block.UPDATE_ALL);
+        int s = (int) Math.ceil((size.getInternalSize() / 2) + 1);
+
+        if (world.getBlockEntity(center) instanceof StructureBlockEntity SBE) {
+            SBE.setMode(StructureMode.LOAD);
+            SBE.setStructureName(MOD_ID + ":" + size.getName());
+            SBE.setStructurePos(new BlockPos(-s, -s, -s));
+            SBE.loadStructure((ServerLevel) world, false);
         }
```

Comment:
You don't need to use a structure block for this, look at how Compact Crafting loads structures on field invalidation (MiniaturizationField) for an example.